catanatron_core/catanatron/models/actions.py

```diff
@@ -51,14 +51,26 @@ def generate_playable_actions(state) -> List[Action]:
     elif action_prompt == ActionPrompt.MOVE_ROBBER:
         return robber_possibilities(state, color)
     elif action_prompt == ActionPrompt.PLAY_TURN:
+        actions = []
+        # Allow playing dev cards before and after rolling
+        if player_can_play_dev(state, color, "YEAR_OF_PLENTY"):
+            actions.extend(year_of_plenty_possibilities(color, state.resource_freqdeck))
+        if player_can_play_dev(state, color, "MONOPOLY"):
+            actions.extend(monopoly_possibilities(color))
+        if player_can_play_dev(state, color, "KNIGHT"):
+            actions.append(Action(color, ActionType.PLAY_KNIGHT_CARD, None))
+        if (
+            player_can_play_dev(state, color, "ROAD_BUILDING")
+            and len(road_building_possibilities(state, color, False)) > 0
+        ):
+            actions.append(Action(color, ActionType.PLAY_ROAD_BUILDING, None))
+
         if state.is_road_building:
             actions = road_building_possibilities(state, color, False)
```

Comment:
In the scenario they are road building, I think I'd like the game state to stay focused on that what do you think? Feels a bit off to: Play Road Building, Place a Road, Play Knight, ... It might lead to subtle bugs of states we are not thinking about.

Could you keep this branch of `state.is_road_building` to only return those options? Thanks!

Reply:
Maybe then the final structure is something like:
```python
if state.is_road_building:
   return road_building_possibilities...
else:
  # Allow playing dev cards before and after rolling
  ...
  if not player_has_rolled(state, color):
    actions.append(ROLL)
   else:
      build houses, roads, cities...
  return actions
```

Reply:
Great catch! Didn't realize this testing thru the UI because is_road_building prevents the UI player from doing anything else